### PR TITLE
Nullable: Interop: Kernel32 and few others

### PR DESCRIPTION
**Pull request by krwq**
couple of easy ones

### CODE REVIEW

**Review by safern — APPROVED** [review]
Thanks! 😄

File: src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs [hunk 1]
```diff
@@ -33,8 +35,9 @@ internal partial class Kernel32
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile)
         {
-            lpFileName = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName);
-            return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, ref securityAttrs, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            string? lpFileNameWithPrefix = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName); // TODO-NULLABLE: API marked as `DO NOT USE`
+            Debug.Assert(lpFileNameWithPrefix != null, "null not expected when non-null passed");
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs]: Please add a `// TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/26761` comment.

File: src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs [hunk 2]
```diff
@@ -33,8 +35,9 @@ internal partial class Kernel32
             int dwFlagsAndAttributes,
             IntPtr hTemplateFile)
         {
-            lpFileName = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName);
-            return CreateFilePrivate(lpFileName, dwDesiredAccess, dwShareMode, ref securityAttrs, dwCreationDisposition, dwFlagsAndAttributes, hTemplateFile);
+            string? lpFileNameWithPrefix = PathInternal.EnsureExtendedPrefixOverMaxPath(lpFileName); // TODO-NULLABLE: API marked as `DO NOT USE`
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs]: This isn't a nullable-related todo.

Comment (stephentoub) [reply src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs]: @JeremyKuhne, we're still using APIs we're not supposed to be using apparently?  Can we either fix the comments on the target method or update the code to not use that method and delete it?

File: src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.FindFirstFileEx.cs [hunk 3]
```diff
@@ -19,10 +21,11 @@ internal partial class Kernel32
 
         internal static SafeFindHandle FindFirstFile(string fileName, ref WIN32_FIND_DATA data)
         {
-            fileName = PathInternal.EnsureExtendedPrefixIfNeeded(fileName);
+            string? fileNameWithPrefix = PathInternal.EnsureExtendedPrefixIfNeeded(fileName);
+            Debug.Assert(fileNameWithPrefix != null, "null not expected when non-null passed");
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.FindFirstFileEx.cs]: `// TODO-NULLABLE: https://github.com/dotnet/roslyn/issues/26761`

File: src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.GetFileAttributesEx.cs [hunk 4]
```diff
@@ -19,8 +21,9 @@ internal partial class Kernel32
 
         internal static bool GetFileAttributesEx(string name, GET_FILEEX_INFO_LEVELS fileInfoLevel, ref WIN32_FILE_ATTRIBUTE_DATA lpFileInformation)
         {
-            name = PathInternal.EnsureExtendedPrefixOverMaxPath(name);
-            return GetFileAttributesExPrivate(name, fileInfoLevel, ref lpFileInformation);
+            string? nameWithExtendedPrefix = PathInternal.EnsureExtendedPrefixOverMaxPath(name); // TODO-NULLABLE: this API is marked as 'DO NOT USE'
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.GetFileAttributesEx.cs]: Same comments.

File: src/System.Private.CoreLib/shared/Interop/Windows/Normaliz/Interop.Normalization.cs [hunk 5]
```diff
@@ -18,7 +19,7 @@ internal partial class Normaliz
                                         string source,
                                         int sourceLength,
                                         [System.Runtime.InteropServices.OutAttribute()]
-                                        char[] destination,
+                                        char[]? destination, // when null, checks the output length
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/Normaliz/Interop.Normalization.cs]: Is this comment needed?

File: src/System.Private.CoreLib/shared/Interop/Windows/OleAut32/Interop.SysAllocStringByteLen.cs [hunk 6]
```diff
@@ -2,14 +2,16 @@
...
 
+#nullable enable
 using System;
 using System.Runtime.InteropServices;
 
 internal partial class Interop
 {
     internal partial class OleAut32
     {
+        // When str is null string is uninitialized
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/OleAut32/Interop.SysAllocStringByteLen.cs]: Is this comment needed?

File: src/System.Private.CoreLib/shared/Interop/Windows/OleAut32/Interop.SysAllocStringLen.cs [hunk 7]
```diff
@@ -13,7 +14,8 @@ internal partial class OleAut32
         [DllImport(Libraries.OleAut32, CharSet = CharSet.Unicode)]
         internal static extern SafeBSTRHandle SysAllocStringLen(IntPtr src, uint len);
 
+        // When str is null string is uninitialized
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/Interop/Windows/OleAut32/Interop.SysAllocStringLen.cs]: Same for all of these.

**Review by stephentoub — APPROVED** [review]
Other than my comments, LGTM.

File: src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs]: Yes, it needs removed, it appears to have gotten lost in the shuffle. The only ones left are here and FindFirstFileEx that I can see.

Comment (krwq) [reply src/System.Private.CoreLib/shared/Interop/Windows/Kernel32/Interop.CreateFile.cs]: replacing with `EnsureExtendedPrefixIfNeeded` (per comment) and removing API marked as `DO NOT USE`

**Review by JeremyKuhne — APPROVED** [review]
Thanks!

**Pull request merged**